schema-engine-tag/src/main/java/org/apache/iotdb/lsm/manager/WALManager.java

```diff
@@ -39,33 +44,145 @@ public abstract class WALManager {
+      IWALRecord walRecord)
       throws IOException {
     this.walDirPath = walDirPath;
-    initFile(walDirPath, walFileName);
-    walWriter = new WALWriter(walFile, walBufferSize, forceEachWrite);
-    walReader = new WALReader(walFile, walRecord);
+    this.walFilePrefix = walFilePrefix;
+    this.flushDirPath = flushDirPath;
+    this.flushFilePrefix = flushFilePrefix;
+    initRecover(walRecord);
     recover = false;
   }
 
-  private void initFile(String walDirPath, String walFileName) throws IOException {
-    File schemaDir = new File(walDirPath);
-    schemaDir.mkdirs();
+  public void initRecover(IWALRecord walRecord) throws IOException {
+    checkPoint();
+    File walDir = new File(walDirPath);
+    walDir.mkdirs();
+    File[] walFiles = walDir.listFiles();
+    walFileNames =
+        Arrays.stream(walFiles)
+            .map(this::getWalFileID)
+            .sorted()
+            .map(this::getWalFileName)
+            .toArray(String[]::new);
+    String walFileName;
+    if (walFileNames.length == 0) {
+      walFileName = getWalFileName(currentFileID);
+    } else {
+      walFileName = walFileNames[currentFileIndex];
+    }
+    initWalWriterAndReader(initFile(walFileName), walRecord);
+  }
+
+  private void checkPoint() {
+    File flushDir = new File(flushDirPath);
+    flushDir.mkdirs();
+    File[] flushFiles = flushDir.listFiles();
+    String[] flushTmpFileNames =
+        Arrays.stream(flushFiles)
+            .map(File::getName)
+            .filter(name -> name.endsWith(SchemaRegionConstant.TMP))
+            .toArray(String[]::new);
+    Integer[] flushTmpIDs =
+        Arrays.stream(flushTmpFileNames).map(this::getFlushFileID).toArray(Integer[]::new);
+    File walDir = new File(walDirPath);
+    walDir.mkdirs();
+    File[] walFiles = walDir.listFiles();
+    Set<Integer> walFileIDs =
+        Arrays.stream(walFiles).map(this::getWalFileID).collect(Collectors.toSet());
+    for (int i = 0; i < flushTmpFileNames.length; i++) {
+      // have tifile tmp don't have wal, tifile is complete, need rename tifile.
+      File flushTmpFile = new File(flushDirPath + File.separator + flushTmpFileNames[i]);
+      String flushFileName =
+          flushTmpFileNames[i].substring(
+              0, flushTmpFileNames[i].length() - SchemaRegionConstant.TMP.length());
+      if (!walFileIDs.contains(flushTmpIDs[i])) {
+        File flushFile = new File(flushDirPath + File.separator + flushFileName);
+        flushTmpFile.renameTo(flushFile);
+      }
+      // have tifile tmp and have wal, tifile is incomplete,delete tifile tmp and delete file.
+      else {
+        flushTmpFile.delete();
+        String flushDeleteFileName =
+            ConvertUtils.getFlushDeleteFileNameFromFlushFileName(flushFileName);
+        File flushDeleteFile = new File(flushDirPath + File.separator + flushDeleteFileName);
+        if (flushDeleteFile.exists()) {
+          flushDeleteFile.delete();
+        }
+      }
+    }
+  }
```

Comment:
Maybe move this to RecoverManager is better. This method processes files more than WAL.